crates/net/downloaders/src/optimistic/error.rs

```diff
@@ -0,0 +1,16 @@
+use reth_primitives::H256;
+use thiserror::Error;
+
+/// The downloader result.
+pub type DownloaderResult<Ok> = std::result::Result<Ok, DownloaderError>;
+
+/// The downloader error.
+#[derive(Debug, Error)]
+pub enum DownloaderError {
+    /// The downloader is at capacity and cannot process new requests.
+    #[error("Downloader is at maximum capacity.")]
+    Full,
```

Comment:
Note here that this should not be an error that reaches the stages as it doesn't matter if the downloader is full, the stages only empty the buffer. Not sure when this error is thrown?